### PR TITLE
Issue-475 : Using the unsigned operations where it s available on PrimitiveType

### PR DESCRIPTION
**Pull request by PapisKang**
Fixes https://github.com/ForNeVeR/Cesium/issues/475

On `PrimitiveType.cs`, `PrimitiveTypeKind.UnsignedShort` and `PrimitiveTypeKind.UnsignedInt` was using signed load operations. 

On this pull request, i ve updated those operations to unsigned methods. 

Base on tests i ve made in C#
Sign-Extension is used when loading a signed value
Zero-Extension when loading an unsigned.

Unsigned types are loaded with ldind.u* operations. 

```
sbyte[] signedBytes = { -1 };
byte[] unsignedBytes = { 255 };

sbyte signedResult = signedBytes[0];
byte unsignedResult = unsignedBytes[0];

int signedInt = (int)signedResult;
int unsignedInt = (int)unsignedResult;
```
image

### CODE REVIEW

File: Cesium.CodeGen/Ir/Types/PrimitiveType.cs [hunk 1]
```diff
@@ -188,9 +188,9 @@ internal static class PrimitiveTypeInfo
...
         { PrimitiveTypeKind.UnsignedChar, (OpCodes.Ldind_U1, OpCodes.Stind_I1) },
         { PrimitiveTypeKind.Short, (OpCodes.Ldind_I2, OpCodes.Stind_I2) },
-        { PrimitiveTypeKind.UnsignedShort, (OpCodes.Ldind_I2, OpCodes.Stind_I2) },
+        { PrimitiveTypeKind.UnsignedShort, (OpCodes.Ldind_U2, OpCodes.Stind_I2) },
         { PrimitiveTypeKind.Int, (OpCodes.Ldind_I4, OpCodes.Stind_I4) },
-        { PrimitiveTypeKind.UnsignedInt, (OpCodes.Ldind_I4, OpCodes.Stind_I4) },
+        { PrimitiveTypeKind.UnsignedInt, (OpCodes.Ldind_U4, OpCodes.Stind_I4) },
         { PrimitiveTypeKind.Float, (OpCodes.Ldind_R4, OpCodes.Stind_R4) },
         { PrimitiveTypeKind.Long, (OpCodes.Ldind_I8, OpCodes.Stind_I8) },
         { PrimitiveTypeKind.UnsignedLong, (OpCodes.Ldind_I8, OpCodes.Stind_I8) },
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Ir/Types/PrimitiveType.cs:196]: There's `Ldind_U8` as well, can you update this one as well? Other than this, LGTM, thanks for fix and investigation!

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

File: Cesium.CodeGen/Ir/Types/PrimitiveType.cs (hunk 1 above)
Comment (BadRyuner) [reply Cesium.CodeGen/Ir/Types/PrimitiveType.cs:196]: ldind_u8 does not exist.

Comment (ForNeVeR) [reply Cesium.CodeGen/Ir/Types/PrimitiveType.cs:196]: Ah, I wasn't attentive enough when reading the instruction list. I double-checked it is present in the list, but missed the fact it's declared as an alias for `ldind.i8`.
image

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**